### PR TITLE
Feat: Added onboarding screen

### PR DESCRIPTION
**Pull request by IbraDajani**
# 🚨 Fill in the fields below

[Create Onboarding screen with all categories](https://trello.com/c/mhQLqcT0/43-create-onboarding-screen-with-all-categories)

## Description

```
~> Created OnBoarding screen with all categories available for selection by the user. and Created source request
```

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] PRODUCT

  - [x] Put PR link on ticket
  - [x] Move the ticket

- [x] DEVELOPERS

  - [x] My code follows the style guidelines of this project
  - [x] I have performed a self-review of my own code
  - [x] I have commented my code, particularly in hard-to-understand areas
  - [x] My changes generate no new warnings and I added `TODO:`, `FIXME:`

- [x] REVIEW
  - [x] Add reviewers: The entire team and most influential people in the area/project
  - [x] Assignees myself to it

🚨 - `Approve: Wait at least one influential in the project or more experienced`

# FRONTEND

## Design
image

## Screenshot or video of your application

https://user-images.githubusercontent.com/75001910/180637605-4e700ec6-7c51-40e1-8b8a-05875e7cbe2d.mp4

### CODE REVIEW

**Review by JeanSilvany — APPROVED** [review]

**Comment by JeanSilvany** [comment]
Good job!

File: src/screens/OnBoarding/index.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import {StatusBar} from 'react-native';
+import Button from '~/components/Button';
+import Divider from '~/components/Divider';
+import RowBetween from '~/components/RowBetween';
+import Separator from '~/components/Separator';
+import Text from '~/components/Text';
```
Comment (PedrozdR) [inline src/screens/OnBoarding/index.tsx:7]: Maybe we can do the same that we did with the App screens. We may organize those components into a index file and export everything from there

File: src/screens/OnBoarding/index.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    <Container>
+      <Content>
+        <StatusBar
+          backgroundColor="transparent"
+          translucent
+          barStyle={handleBarStyle}
+        />
+        <RowBetween>
+          <Text color="text">
+            App<Text color="primary">News</Text>
```
Comment (PedrozdR) [inline src/screens/OnBoarding/index.tsx:44]: Lets try to put this header text with other typography.

File: src/screens/OnBoarding/index.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+              }
+              onPress={handleCategoriesPick(category)}
+            >
+              <CategoryLabel
+                isSelected={
+                  selectedCategories[category.label] === category.label
+                }
+              >
+                {category.label}
+              </CategoryLabel>
```
Comment (PedrozdR) [inline src/screens/OnBoarding/index.tsx:72]: We need to use some emojis following this label. The shortcut is Windows + . (dot) 👳🏻‍♂️

Comment (PedrozdR) [reply src/screens/OnBoarding/index.tsx:72]: And try to put these Categories with rounded containers, like circles. 
Following the layout below 
image

File: src/screens/OnBoarding/index.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+          backgroundColor="transparent"
+          translucent
+          barStyle={handleBarStyle}
+        />
+        <RowBetween>
+          <Text color="text">
+            App<Text color="primary">News</Text>
+          </Text>
+          <SkipButton onPress={handleNavigateToHomeScreen}>
+            <Text typography="overline">Skip</Text>
```
Comment (PedrozdR) [inline src/screens/OnBoarding/index.tsx:47]: The "skip" text button of the layout is with grey color. 
```suggestion
            <Text color={colors.grey} typography="overline">Skip</Text>
```

image

File: src/screens/OnBoarding/useOnboardingController.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+  /**
+   * Callbacks
+   */
+
+  const handleNavigateToHomeScreen = () => {
+    navigate('HomeScreen');
+  };
+
+  const handleCategoriesPick = (category: CategoriesProps) => () => {
```
Comment (PedrozdR) [inline src/screens/OnBoarding/useOnboardingController.tsx:35]: May you explain this function for me please? kkkkk Why you are not using an array of string insted of a object?

**Review by PedrozdR — CHANGES_REQUESTED** [review]

File: src/screens/OnBoarding/index.tsx (hunk 2 above)
Comment (PedrozdR) [reply src/screens/OnBoarding/index.tsx:44]: Like a bold one

File: src/screens/OnBoarding/index.tsx (hunk 1 above)
Comment (IbraDajani) [reply src/screens/OnBoarding/index.tsx:7]: Lets create a card for it

File: src/screens/OnBoarding/index.tsx (hunk 2 above)
Comment (IbraDajani) [reply src/screens/OnBoarding/index.tsx:44]: Agreed

File: src/screens/OnBoarding/useOnboardingController.tsx (hunk 5 above)
Comment (IbraDajani) [reply src/screens/OnBoarding/useOnboardingController.tsx:35]: Is easier to deal with add/remove items from an object than it is on an array...

that said, the function is simple

first, I check if the item the user pressed already exists in the object. if it exists, i create a copy of the whole object, delete the item and then setState the copy of the object, but this time, with the item removed. if it doens't exists, i just add it to the Object


like

item exists ? (remove it) : (add it)

File: src/screens/OnBoarding/index.tsx (hunk 2 above)
Comment (PedrozdR) [reply src/screens/OnBoarding/index.tsx:44]: Both

**Review by PedrozdR — APPROVED** [review]

**Pull request merged**